libs/fairness/src/lib/util/calculateFairnessMetric.test.ts

```diff
@@ -0,0 +1,459 @@
+    const mockValue = {
+      binBounds: [
+        { lower: 0.25, upper: 0.35 },
+        { lower: 0.65, upper: 0.75 }
+      ],
+      bins: [0.3, 0.7],
+      global: 0.5
+    };
+    const result = calculateFairnessMetric(mockValue, FairnessModes.Difference);
+    const expectedResult = {
+      bounds: { lower: closeTo(0.3), upper: closeTo(0.5) },
+      overall: closeTo(0.4)
+    };
+    expect(result).toMatchObject(expectedResult);
+  });
+  it("should calculate difference (without overlap and reverse order) correctly", () => {
+    const mockValue = {
+      binBounds: [
+        { lower: 0.65, upper: 0.75 },
+        { lower: 0.25, upper: 0.35 }
+      ],
+      bins: [0.7, 0.3],
+      global: 0.5
+    };
+    const result = calculateFairnessMetric(mockValue, FairnessModes.Difference);
+    const expectedResult = {
+      bounds: { lower: closeTo(0.3), upper: closeTo(0.5) },
+      overall: closeTo(0.4)
+    };
+    expect(result).toMatchObject(expectedResult);
+  });
+  it("should calculate difference with 3+ groups correctly", () => {
+    const mockValue = {
+      binBounds: [
+        { lower: 0.25, upper: 0.35 },
+        { lower: 0.65, upper: 0.75 },
+        { lower: 0.45, upper: 0.55 }
+      ],
+      bins: [0.3, 0.7, 0.5],
+      global: 0.5
+    };
+    const result = calculateFairnessMetric(mockValue, FairnessModes.Difference);
+    const expectedResult = {
+      bounds: { lower: closeTo(0.1), upper: closeTo(0.5) },
+      overall: closeTo(0.4)
+    };
+    expect(result).toMatchObject(expectedResult);
+  });
+  it("should calculate difference with 3+ groups correctly", () => {
+    const mockValue = {
+      binBounds: [
+        { lower: 0.45, upper: 0.55 },
+        { lower: 0.25, upper: 0.35 },
+        { lower: 0.65, upper: 0.75 }
+      ],
+      bins: [0.5, 0.3, 0.7],
+      global: 0.5
+    };
+    const result = calculateFairnessMetric(mockValue, FairnessModes.Difference);
+    const expectedResult = {
+      bounds: { lower: closeTo(0.1), upper: closeTo(0.5) },
+      overall: closeTo(0.4)
+    };
+    expect(result).toMatchObject(expectedResult);
+  });
+  it("should calculate difference with 3+ groups correctly (with overlap)", () => {
+    const mockValue = {
+      binBounds: [
+        { lower: 0.45, upper: 0.7 },
+        { lower: 0.25, upper: 0.35 },
+        { lower: 0.65, upper: 0.75 }
+      ],
+      bins: [0.5, 0.3, 0.7],
+      global: 0.5
+    };
+    const result = calculateFairnessMetric(mockValue, FairnessModes.Difference);
+    const expectedResult = {
+      bounds: { lower: closeTo(0), upper: closeTo(0.5) },
+      overall: closeTo(0.4)
+    };
+    expect(result).toMatchObject(expectedResult);
+  });
+  it("should handle difference case with large binBound that provides the largest and smallest bounds", () => {
+    const mockValue = {
+      binBounds: [
+        { lower: 0.1, upper: 0.3 },
+        { lower: 0.7, upper: 0.9 },
+        { lower: 0.08, upper: 0.95 }
```

Comment:
This is subtler.... and I haven't quite figured out the answer yet.

However, I would argue that you can only get to a minimum difference of zero for cases like:
```json
        { lower: 0.1, upper: 0.3 },
        { lower: 0.7, upper: 0.9 },
        { lower: 0.08, upper: 0.95 },
        { lower: 0.08, upper: 0.95 }
```
where the repetition allows the max and mins to fully overlap

Reply:
I believe the case should be when all bounds overlap